org.opentosca.bus.management.service.impl/src/org/opentosca/bus/management/service/impl/collaboration/RequestReceiver.java

```diff
@@ -0,0 +1,583 @@
+
+        // reply is only possible if all headers are set
+        if (broker != null && replyTopic != null && correlation != null) {
+
+            // add the header fields to the header map and return it
+            final Map<String, Object> headers = new HashMap<>();
+            headers.put(MBHeader.MQTTBROKERHOSTNAME_STRING.toString(), broker);
+            headers.put(MBHeader.MQTTTOPIC_STRING.toString(), replyTopic);
+            headers.put(MBHeader.CORRELATIONID_STRING.toString(), correlation);
+
+            return headers;
+        } else {
+            // header fields are missing and therefore no response possible
+            return null;
+        }
+    }
+
+    /**
+     * Check whether the request is directed to the local OpenTOSCA Container / Management Bus. This
+     * is the case if the {@link MBHeader#DEPLOYMENTLOCATION_STRING} header field equals the local
+     * host name.
+     *
+     * @param message the request message
+     * @return <tt>true</tt> if the request is directed to this Management Bus, <tt>false</tt>
+     *         otherwise
+     */
+    private boolean isDestinationLocal(final Message message) {
+
+        final String deploymentLocation =
+            message.getHeader(MBHeader.DEPLOYMENTLOCATION_STRING.toString(), String.class);
+        RequestReceiver.LOG.debug("Deplyoment location header: {}", deploymentLocation);
+
+        return deploymentLocation != null && deploymentLocation.equals(Settings.OPENTOSCA_CONTAINER_HOSTNAME);
+    }
+
+    /**
+     * Create a String that uniquely identifies the IA that has to be deployed/undeployed for the
+     * given request message. The String can be used to synchronize all operations that are
+     * concerned with that IA.
+     *
+     * @param message the request message
+     * @return a String that uniquely identifies the IA or <tt>null</tt> if needed header fields are
+     *         missing
+     */
+    private String getUniqueSynchronizationString(final Message message) {
+
+        final String triggeringContainer =
+            message.getHeader(MBHeader.TRIGGERINGCONTAINER_STRING.toString(), String.class);
+        final String deploymentLocation = Settings.OPENTOSCA_CONTAINER_HOSTNAME;
+        final QName typeImplementationID =
+            message.getHeader(MBHeader.TYPEIMPLEMENTATIONID_QNAME.toString(), QName.class);
+        final String implementationArtifactName =
+            message.getHeader(MBHeader.IMPLEMENTATIONARTIFACTNAME_STRING.toString(), String.class);
+
+        if (triggeringContainer != null && deploymentLocation != null && typeImplementationID != null
+            && implementationArtifactName != null) {
+
+            return triggeringContainer + "/" + deploymentLocation + "/" + typeImplementationID.toString() + "/"
+                + implementationArtifactName;
+        } else {
+            return null;
+        }
```

Comment:
inverting the condition improves readability. Using `Stream.of` and `Collectors.joining('/')` can simplify the return value